crates/openvino/src/element_type.rs

```diff
@@ -18,96 +7,126 @@ use std::fmt;
 #[repr(u32)]
 pub enum ElementType {
     /// An undefined element type.
-    Undefined = ov_element_type_e_UNDEFINED,
+    Undefined,
     /// A dynamic element type.
-    Dynamic = ov_element_type_e_DYNAMIC,
+    Dynamic,
     /// A boolean element type.
-    Boolean = ov_element_type_e_OV_BOOLEAN,
+    Boolean,
     /// A Bf16 element type.
-    Bf16 = ov_element_type_e_BF16,
+    Bf16,
     /// A F16 element type.
-    F16 = ov_element_type_e_F16,
+    F16,
     /// A F32 element type.
-    F32 = ov_element_type_e_F32,
+    F32,
     /// A F64 element type.
-    F64 = ov_element_type_e_F64,
+    F64,
     /// A 4-bit integer element type.
-    I4 = ov_element_type_e_I4,
+    I4,
     /// An 8-bit integer element type.
-    I8 = ov_element_type_e_I8,
+    I8,
     /// A 16-bit integer element type.
-    I16 = ov_element_type_e_I16,
+    I16,
     /// A 32-bit integer element type.
-    I32 = ov_element_type_e_I32,
+    I32,
     /// A 64-bit integer element type.
-    I64 = ov_element_type_e_I64,
-    /// An 1-bit unsigned integer element type.
-    U1 = ov_element_type_e_U1,
-    /// An 4-bit unsigned integer element type.
-    U4 = ov_element_type_e_U4,
+    I64,
+    /// A 1-bit unsigned integer element type.
+    U1,
+    /// A 2-bit unsigned integer element type.
+    U2,
+    /// A 3-bit unsigned integer element type.
+    U3,
+    /// A 4-bit unsigned integer element type.
+    U4,
+    /// A 6-bit unsigned integer element type.
+    U6,
     /// An 8-bit unsigned integer element type.
-    U8 = ov_element_type_e_U8,
+    U8,
     /// A 16-bit unsigned integer element type.
-    U16 = ov_element_type_e_U16,
+    U16,
     /// A 32-bit unsigned integer element type.
-    U32 = ov_element_type_e_U32,
+    U32,
     /// A 64-bit unsigned integer element type.
-    U64 = ov_element_type_e_U64,
+    U64,
     /// NF4 element type.
-    NF4 = ov_element_type_e_NF4,
+    NF4,
     /// F8E4M3 element type.
-    F8E4M3 = ov_element_type_e_F8E4M3,
+    F8E4M3,
     /// F8E5M3 element type.
-    F8E5M3 = ov_element_type_e_F8E5M3,
-}
-
-/// Error returned when attempting to create an [`ElementType`] from an illegal `u32` value.
-#[derive(Debug)]
-pub struct IllegalValueError(u32);
-
-impl Error for IllegalValueError {}
-
-impl fmt::Display for IllegalValueError {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "illegal value: {}", self.0)
-    }
+    F8E5M3,
+    /// A string element type.
+    String,
```

Comment:
Exposing `String` without c-api support in this version would be problematic. Might need to add rule in codegen.rs to remove this, or check every function that uses element_type.

Reply:
This is exposed in the C API: https://github.com/intel/openvino-rs/pull/144/files#diff-8098561b920102e40c14aeb6333d026f05b38a5dc4590f281b713aea588a15afR96.

Reply:
They're exposed, but not supported in all APIs yet
https://github.com/openvinotoolkit/openvino/issues/26906#issuecomment-2398517634 
Can't create a tensor with string data, can only create an empty string type tensor for now.

Reply:
That's fine: the job of these bindings is to faithfully pass on the programmer intent, even if the underlying framework has bugs or holes. The issue you refer to shouldn't be resolved here.